code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -49,6 +51,24 @@
 		desc += " It is painted with [department] colors."
 
 	update_icon()
+	var/obj/machinery/door/airlock/newtarget = (locate(/obj/machinery/door/airlock) in get_turf(src))
+	if(newtarget)
+		var/direction = reverse_direction(dir)
+		forceMove(get_step(newtarget.loc, reverse_direction(direction)))
+		if (locate(/obj/machinery/door/airlock) in oview(1, newtarget))
```

Comment:
Why not condense this to a singular `for (var/obj/machinery/door/airlock/A in oview(1, newtarget))` and check the target direction relative to `newtarget` instead?